util/src/main/java/org/datacommons/util/StatChecker.java

```diff
@@ -185,6 +188,70 @@ public synchronized void check() {
     }
   }
 
+  // This feature is not within check() because it uses ExistenceChecker instead
+  // of using the StatChecker.check() mechanism of counter keys.
```

Comment:
Hmm, actually how about calling this from check()?  It is actually checking stats (e.g., the values for measurementResult StatVarObs).

Reply:
The main complication I am avoiding is having to pass `args` (or `args.checkMeasurementResult`) down to StatChecker. I realize I did not communicate this in the comment. I felt that it is better if the concern of when we execute this behavior is isolated from the `StatChecker` module.

I am curious what you think about this

Reply:
If we designate StatChecker the responsibility of checking measurementResult type of stats, then I feel it makes sense to pass whatever is needed for the checking to StatChecker at construction.  So the caller (Processor) does not have to explicitly call into StatChecker again for just validating measurementResult.

Reply:
And we do all the looping once...

Reply:
that makes a lot of sense, thank you for the guidance. implemented.